src/main/java/org/example/dao/PointDAOImpl.java

```diff
@@ -0,0 +1,95 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class PointDAOImpl implements PointDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public PointDAOImpl() {
+    }
+
+    @Override
+    public void insertPoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (PreparedStatement statement = connection.prepareStatement("INSERT INTO points (id, xCoordinate, yCoordinate) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS)) {
+            statement.setInt(1, point.getId());
```

Comment:
We can get rid of this line since we're setting the ID in the while loop below using point.setID(id)